### ISSUE TITLE
Redo kubernetes-public global ip terraform to use for_each

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/k8s.io#2275]
Followup to https://github.com/kubernetes/k8s.io/pull/2242

I would like us to start avoiding so much copy-pasting of terraform, but am opening this up as a followup issue to unblock the PR

**Comment by spiffxp** [comment]
/milestone v1.23
Mostly what's held me back from doing this is figuring out how to test/script the `terraform state mv` stuff without updating live state
